site-src/geps/gep-922.md

```diff
@@ -0,0 +1,163 @@
+# GEP-922: Gateway API Versioning
+
+* Issue: [#922](https://github.com/kubernetes-sigs/gateway-api/issues/922)
+* Status: Implementable
+
+## TLDR
+Each Gateway API release will be represented by a bundle version that represents
+that specific combination of CRDs, API versions, and validating webhook. To
+enable experimental fields, a new experimental set of CRDs will be included with
+future releases.
+
+## Goals
+Provide a path for implementations to support:
+
+* Users should have a predictable experience when interacting with Gateway API. 
+* The API should be able to continue to safely evolve, including:
+    * Loosening validation
+    * Transitioning required fields to optional
+    * Adding new fields at an alpha stability level
+
+## Introduction
+This GEP details how we should approach future releases of Gateway API. This is
+intended as another step towards building consensus. Once merged, it will be
+considered implemented once this content is fully documented.
+
+## Version Terminology
+Gateway API has two related but different versioning schemes:
+
+### 1. API Versions (ex: v1alpha2)
+Each API version provides a unique way to interact with the API. Significant
+changes such as removing or renaming fields will be represented here.
+
+### 2. Bundle Versions (ex: v0.4.0)
+Bundle versions are broader in scope. They use semantic versioning and include
+the following:
+
+* API Types/CRDs
+* Validating Webhook
+
+Each bundle may include multiple API versions, potentially introducing new ones
+and/or removing old ones as part of a new bundle.
+
+## Limitations of Webhook and CRD Validation
+CRD and webhook validation is not the final validation i.e. webhook is “nice UX”
+but not schema enforcement. This validation is intended to provide immediate
+feedback to users when they provide an invalid configuration, but can not
+completely be relied on because it:
+
+* Is not guaranteed to be present or up to date in all clusters.
+* Will likely never be sufficient to cover all edge cases.
+* May be loosened in future API releases.
+
+## Role Requirements
+When implementing or using Gateway API, each role has a unique set of
+requirements to ensure we're providing a consistent experience.
+
+### API Authors:
+* MUST support API versions for at least 1 year, starting with v1alpha2.
+* MUST provide conversion between API versions, starting with v1alpha2.
```

Comment:
To be clear, this doesn't imply conversion webhook, correct?
Does this imply round-trip to lower versions is required?

Reply:
As far as I can tell, a conversion webhook is necessary for the following:

1. Renaming fields
2. Restructuring fields

I'd like to avoid a conversion webhook as long as possible, but I do think it's likely preferable to having incompatible API versions, especially if we're expecting our next release to be v1beta1.